src/data/list/basic.lean

```diff
@@ -1941,6 +1941,38 @@ begin
     apply hl _ (list.mem_cons_self _ _) }
 end
 
+/-- Transform a `l : list α` folded via `foldr` across a motive that holds
+for the seed element `b : β` and for all incremental `op : α → β → β`
+performed on the elements `(a : α) ∈ l`. -/
+def foldr_induction {C : β → Sort*} (l : list α) (op : α → β → β) (b : β)
```

Comment:
Since this operators on `Sort`, should this be called `_rec` instead of `induction`?